connection-cache/src/connection_cache.rs

```diff
@@ -156,6 +197,60 @@ where
         }
     }
 
+    fn create_connection_internal(
+        config: Arc<C>,
+        connection_manager: Arc<M>,
+        map: &mut std::sync::RwLockWriteGuard<'_, IndexMap<SocketAddr, P>>,
+        addr: &SocketAddr,
+        connection_pool_size: usize,
+        async_connection_sender: Option<Sender<(usize, SocketAddr)>>,
```

Comment:
Is it possible to change this to only take references to the underlying types rather than cloning all the arcs and sender?